### ISSUE TITLE
Image.open with non-image file raises IOError without closing the file

### ISSUE THREAD

**Issue by ZMI-JacobGeiman** [python-pillow/Pillow#1734]
Attempting to open a non-image file and catching the IOError causes the file handle to be left open, meaning this file cannot be modified by the process/os until the application stops.

Example code:
```
from PIL import Image
import os
path = '/path/to/non/image'
try:
    Image.open(path)
except IOError as err:
    print 'This is not an image file'
os.remove(path)
```
I am running Pillow 3.1.1 on Python 2.7.6
This should be fixable by calling fp.close() before raising the IOError in Image.open

**Comment by radarhere** [comment]
Just to confirm, you're running Windows, yes?

**Comment by ZMI-JacobGeiman** [comment]
yes, this was on Windows 7.

**Comment by radarhere** [comment]
Okay then. Do you mind if this issue is closed as a duplicate then? Between #1651 and #1144, I believe that this is covered.

**Comment by ZMI-JacobGeiman** [comment]
please close. Thanks
